src/app_service/service/community/service.nim

```diff
@@ -1601,13 +1602,22 @@ QtObject:
     self.events.emit(SIGNAL_COMMUNITY_DATA_IMPORTED, CommunityArgs(community: community))
     self.events.emit(SIGNAL_COMMUNITIES_UPDATE, CommunitiesArgs(communities: @[community]))
 
+  proc getSharedAddresses*(self: Service): seq[string] =
+    return self.sharedAddresses
+
   proc asyncCheckPermissionsToJoin*(self: Service, communityId: string, addresses: seq[string]) =
+
+    if addresses.len == 0:
+      debug "Addresses received cannot be empty"
```

Comment:
That's not true currently. If the array is empty, it then means status-go will check all addresses.

When we are already in a community, yes, we should usually pass the list of already selected addresses. However, this function is also used when editing the shared addresses. So it validates if the newly selected addresses still work.

There are two options: 
1. leave it as it was. This has the downside of maybe reintroducing the bug you're fixing
2. Update it to have to always pass addresses. It's more work, but it does make it more explicit. You'll need to update the QML to send all addresses when the user does not manually select addresses.

Reply:
So the `CommunityMembershipSetupDialog` gets the accounts from `walletAccountsModel`
I guess I need to fetch those addresses from that model, and propagate the addresses with a sort filter proxy model to the chat section so that it can be transferred to the status_go.

Reply:
For the Edit Shared Addresses popup, we should already be doing that, though there seems to be a bug, so maybe there was a regression.

Reply:
Yes, I had checked the flow for Edit shared addresses, we do pass the addresses.

We're missing sharing the addresses when setting the item active in a chat section.

Small callout, is it normal that we recalculate the permissions when just selecting/deselecting the accounts in the `Edit Shared addresses` popup ?

Reply:
yes. The goal is to see if the new addresses you selected fulfill the conditions
It lets you know which combination lets you see all channels for example